Seeding the initial state of the environment

skim0119 opened the pull request:
# Issue
For debugging purpose, it would be nice to be able to fix the environment. The env.seed() method is the most intuitive method to use, but it is not functioning correctly.

# Solution
I think the current issue is that the env.seed() method only fix the seed of the random generator class within the gym environment, and it does not fix the numpy seed. I simply added a line that fix the numpy seed, and it would be a temporary solution.

``` py
...
self.np_random, seed = seeding.np_random(seed)
np.random.seed(seed)
return [seed]
```

# For Future
The problem with this code is that all the numpy random generator would be fixed. If we want to use numpy random somewhere else with fixed environment, it would cause some problems. Ideally, all the random number generation in initializing environment needs to use the separate random generator with its own seed.

skim0119 committed:
Attempt to fix the environment

For debugging purpose, it would be nice to fix the environment. The current issue is the env.seed() method only fix the random generator class within the gym environment, and it does not fix the numpy seed. I added a line that fix the numpy seed, and it would be a temporary solution.
The problem with this code is that all the numpy random generator would be fixed. If we want to use numpy random somewhere else along with fixed environment, it would cause some problems. Ideally, all the random number generation in initializing environment needs to use the random generator class with its own seed.

osipychev commented:
The seeding is done in a similar fashion to other gym environments (take a look at box2d and blackjack). Env.seed dos not require np.random.seed.

The proper way of fixing the seeding is to create a random generator instance (self.np_random) and use it all along the environment. This way the environment will be using an independent instance from the control random generator. 
If you do it the way you proposed, you will use the same np.random instance as in the controller script that will affect the random draws in environment.

skim0119 committed:
change np.random to np_random generator

skim0119 committed:
Update create_map.py

skim0119 committed:
pass np_random to CreateMap.gen_map method

skim0119 commented:
As you mentioned, I modified the code to use the random generator instance (self.np_random) when it resets the map. I checked the map is correctly fixed by the method env.seed.

# Changes
- Removed the np.random.seed line from previous commits.
- On static method CreateMap.gen_map, I added np_random parameter to pass the random generator instance.
- Changed np.random in gen_map to np_random instead.
